pkg/restore/restore.go

```diff
@@ -198,42 +198,85 @@ func (kr *kubernetesRestorer) Restore(restore *api.Restore, backup *api.Backup,
 		return api.RestoreResult{}, api.RestoreResult{Ark: []string{err.Error()}}
 	}
 
-	dir, err := kr.unzipAndExtractBackup(backupReader)
+	gzippedLog := gzip.NewWriter(logFile)
+	defer gzippedLog.Close()
+
+	ctx := &context{
+		backup:               backup,
+		backupReader:         backupReader,
+		restore:              restore,
+		prioritizedResources: prioritizedResources,
+		selector:             selector,
+		logger:               &logger{w: gzippedLog},
+		dynamicFactory:       kr.dynamicFactory,
+		fileSystem:           kr.fileSystem,
+		namespaceClient:      kr.namespaceClient,
+		restorers:            kr.restorers,
+	}
+
+	return ctx.execute()
+}
+
+type logger struct {
+	w io.Writer
+}
+
+func (l *logger) log(msg string, args ...interface{}) {
+	// TODO use a real logger that supports writing to files
+	now := time.Now().Format(time.RFC3339)
+	fmt.Fprintf(l.w, now+" "+msg+"\n", args...)
+}
+
+type context struct {
+	backup               *api.Backup
+	backupReader         io.Reader
+	restore              *api.Restore
+	prioritizedResources []schema.GroupResource
+	selector             labels.Selector
+	logger               *logger
+	dynamicFactory       client.DynamicFactory
```

Comment:
it seems a little weird to me to have things like the dynamicFactory, fs, nsClient to be part of the context. I'm also not sold yet on moving all of the methods to have context receivers -- I thought context was going to be passed as an arg. I think it's fine for now as-is but we should re-visit the overall backup/restore design for 0.5

Reply:
Maybe context isn't the right word. It's really "here's a whole bunch of stuff we need to do a backup/restore and I don't really want to keep passing the same 6 arguments to every function over and over." If you're ok with it for now, we can and should definitely revisit for 0.5.

Reply:
yep, WFM.